Documentation/rust/unit-testing.rst

```diff
@@ -0,0 +1,49 @@
+.. SPDX-License-Identifier: GPL-2.0
+
+Unit Testing
+============
+
+Like in any other Rust project it is possible to write and run unit tests and
+documentation tests in the kernel.
+
+Running Unit Tests
+------------------
+
+Unit tests in the kernel are identical to user-space Rust tests:
+
+.. code-block:: rust
+
+	#[cfg(test)]
+	mod tests {
+	    #[test]
+	    fn it_works() {
+	        let result = 2 + 2;
+	        assert_eq!(result, 4);
+	    }
+	}
+
+And can be run using the ``rusttest`` Make target:
+
+.. code-block:: bash
+
+	$ make LLVM=1 rusttest
+
+Running Documentation Tests
+---------------------------
+
+Like in user-space, it is possible to write documentation tests:
+
+.. code-block:: rust
+
+	/// ```
+	/// let result = 2 + 2;
+	/// assert_eq!(result, 4);
+	/// ```
+
+Documentation tests use KUnit and, thanks to the configuration file present in
+``rust/.kunitconfig``, it is possible to run them using the ``kunit.py`` tool:
```

Comment:
I was quoting the sentence in the docs, not the file; i.e. I didn't mean to remove the file! :)

What I meant is that the sentence above says "thanks to ..." when it is not really due to the file (and it is not even required), so it can be misleading.

I think the docs could say here something like: "Documentation tests use KUnit, and therefore can be run either on boot or via the ``kunit.py`` tool: ...", and probably also link the KUnit docs etc.

Reply:
v2 rephrased this sentence to clarify that `rust/.kunitconfig` is not needed, just convinient.